script/upgrades/eXOF/eXOFChecks.verify.sol

```diff
@@ -0,0 +1,517 @@
+    require(
+      exchanges.length == config.pools.length + 7,
+      "Number of expected pools does not match the number of deployed pools."
+    );
+
+    for (uint256 i = 0; i < config.pools.length; i++) {
+      bytes32 exchangeId = getExchangeId(config.pools[i].asset0, config.pools[i].asset1, config.pools[i].isConstantSum);
+
+      IBiPoolManager.PoolExchange memory pool = BiPoolManager(biPoolManagerProxy).getPoolExchange(exchangeId);
+      Config.Pool memory poolConfig = config.pools[i];
+
+      // verify asset0 of the deployed pool against the config
+      if (pool.asset0 != poolConfig.asset0) {
+        console.log(
+          "The asset0 of deployed pool: %s does not match the expected asset0: %s.",
+          pool.asset0,
+          poolConfig.asset0
+        );
+        revert("asset0 of pool does not match the expected asset0. See logs.");
+      }
+
+      // verify asset1 of the deployed pool against the config
+      if (pool.asset1 != poolConfig.asset1) {
+        console.log(
+          "The asset1 of deployed pool: %s does not match the expected asset1: %s.",
+          pool.asset1,
+          poolConfig.asset1
+        );
+        revert("asset1 of pool does not match the expected asset1. See logs.");
+      }
+
+      if (poolConfig.isConstantSum) {
+        if (address(pool.pricingModule) != constantSum) {
+          console.log(
+            "The pricing module of deployed pool: %s does not match the expected pricing module: %s.",
+            address(pool.pricingModule),
+            constantSum
+          );
+          revert("pricing module of pool does not match the expected pricing module. See logs.");
+        }
+      } else {
+        if (address(pool.pricingModule) != constantProduct) {
+          console.log(
+            "The pricing module of deployed pool: %s does not match the expected pricing module: %s.",
+            address(pool.pricingModule),
+            constantProduct
+          );
+          revert("pricing module of pool does not match the expected pricing module. See logs.");
+        }
+      }
+      // verify asset0 is always a stable asset
+      require(
+        pool.asset0 == cUSD || pool.asset0 == cEUR || pool.asset0 == cBRL || pool.asset0 == eXOF,
+        "asset0 is not a stable asset in the exchange"
+      );
+      // verify asset1 is always a collateral asset
+      require(
+        pool.asset1 == celoToken || pool.asset1 == bridgedUSDC || pool.asset1 == bridgedEUROC,
+        "asset1 is not CELO or bridgedUSDC in the exchange"
+      );
+    }
+    console.log("🟢 PoolExchange correctly configured 🤘🏼");
+  }
+
+  function verifyPoolConfig(eXOFConfig.eXOF memory config) internal view {
+    for (uint256 i = 0; i < config.pools.length; i++) {
+      bytes32 exchangeId = getExchangeId(config.pools[i].asset0, config.pools[i].asset1, config.pools[i].isConstantSum);
+      IBiPoolManager.PoolExchange memory pool = BiPoolManager(biPoolManagerProxy).getPoolExchange(exchangeId);
```

Comment:
maybe something like `deployedPool` or `actualPool` would be a better name here, otherwise the comparison look a bit confusing (i.e. `pool.config.spread.unwrap() != poolConfig.spread.unwrap()`). Similarly in all the other functions that check the actual pools vs the pool configuration.

Reply:
The naming choice is definitely poor, however reading the code where the variables are populated would clarify any confusion.